WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -3463,6 +3467,35 @@ public void onTrackableEvent(TrackableEvent event, Map<String, String> propertie
         AppPrefs.setGutenbergBlockTypeImpressions(impressions);
     }
 
+    @Override public void onContactCustomerSupport() {
+        // construct a mutable list to add the related and extra tags
+        ArrayList<String> tagsList = new ArrayList<>();
+
+        // Append the "mobile_gutenberg_is_default" tag if gutenberg is set to default for new posts
+        if (SiteUtils.isBlockEditorDefaultForNewPost(getSite())) {
+            tagsList.add(ZendeskExtraTags.gutenbergIsDefault);
+        }
+
+        mZendeskHelper.createNewTicket(
+                this,
+                Origin.EDITOR_HELP,
+                getSite(),
+                tagsList.isEmpty() ? null : tagsList
+        );
+    }
+
+    @Override public void onGotoCustomerSupportOptions() {
+        // construct a mutable list to add the related and extra tags
+        ArrayList<String> tagsList = new ArrayList<>();
+
+        // Append the "mobile_gutenberg_is_default" tag if gutenberg is set to default for new posts
+        if (SiteUtils.isBlockEditorDefaultForNewPost(getSite())) {
+            tagsList.add(ZendeskExtraTags.gutenbergIsDefault);
+        }
+
+        ActivityLauncher.viewHelpAndSupport(this, Origin.EDITOR_HELP, getSite(), tagsList.isEmpty() ? null : tagsList);
+    }
```

Comment:
Just a personal preference, but if I were doing this I would move this code out to some kind of helper class/object to avoid growing `EditPostActivity` more than necessary. Also, it would let me write this in Kotlin (not a small part of the reason I would move it 😉 ).

I was just curious to see what that might look like, and one relatively simple possibility (avoiding any DI complexity) would be something like this:

```
object EditPostCustomerSupportHelper {
    fun onContactCustomerSupport(zendeskHelper: ZendeskHelper, context: Context, site: SiteModel) {
        zendeskHelper.createNewTicket(context, EDITOR_HELP, site, getTagsList(site))
    }

    fun onGotoCustomerSupportOptions(context: Context, site: SiteModel) {
        ActivityLauncher.viewHelpAndSupport(context, EDITOR_HELP, site, getTagsList(site))
    }

    private fun getTagsList(site: SiteModel): List<String>? =
        if (SiteUtils.isBlockEditorDefaultForNewPost(site)) {
            listOf(ZendeskExtraTags.gutenbergIsDefault)
        } else {
            null
        }
}
```

Reply:
I _was_ feeling dirty when I noticed I was adding new lines in EditPostActivity 😬. Good idea about extracting to a separate module Matt and thanks for the code! Addressed with d5abf4ce710135c6f70b7237c64cea175312dd99.